sdk/java/src/main/java/com/chain/http/Client.java

```diff
@@ -141,13 +143,15 @@ public Client(URL url, String accessToken) {
    * @return the result of the post request
    * @throws ChainException
    */
-  public <T> BatchResponse<T> batchRequest(String action, Object body, Type tClass, Type eClass)
-      throws ChainException {
-    return post(
-        action,
-        body,
-        (Response response, Gson deserializer) ->
-            new BatchResponse(response, deserializer, tClass, eClass));
+  public <T> BatchResponse<T> batchRequest(
+      String action, Object body, final Type tClass, final Type eClass) throws ChainException {
+    ResponseCreator<T> rc =
+        new ResponseCreator<T>() {
+          public T create(Response response, Gson deserializer) throws ChainException, IOException {
+            return (T) new BatchResponse<T>(response, deserializer, tClass, eClass);
```

Comment:
This does work at runtime, but your suggested changes add clarity to the code. I'll update.